WordPress/openverse
# Update deployment doc and add zero-downtime info about environment variables

## Fixes




Fixes #876  by @sarayourfriend 

## Description



Updates the existing frontend deployment documentation to be a general "ECS deployment" document. In updating this, I also realised that environment variables were not sufficiently documented, particularly with respect to the edge cases around zero-downtime deployments and rollbacks. I've gone ahead and also added a new section to the zero-downtime deployments document to account for this.

## Testing Instructions


Read the updated documents and ensure they make sense and can be understood.

You can view the new deployment document in the documentation preview here: https://wordpress.github.io/openverse/_preview/907/guides/deployment.html

## Checklist



      `Update index.md`).
      a parent feature branch.
- [N/A] I added or updated tests for the changes I made (if applicable).
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
